From 865226c3f31b1d295bda429037245c09b378b6e7 Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Mon, 25 Jan 2016 18:08:00 +0800
Subject: [PATCH 0737/1110] drm/amd/dal: remove dal_services.h

Also rename dal_services_types.h to dc_services_types.h
Fix dependent components and DMs

Signed-off-by: Mykola Lysenko <Mykola.Lysenko@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
---
 .../drm/amd/dal/amdgpu_dm/amdgpu_dal_services.c    |   2 +-
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c      |   2 +-
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_irq.c  |   2 +-
 .../drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c    |   2 +-
 .../gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c    |   2 +-
 .../gpu/drm/amd/dal/dal_power_interface_types.h    |  76 -------
 .../gpu/drm/amd/dal/dc/adapter/adapter_service.c   |   2 +-
 .../adapter/dce110/hw_ctx_adapter_service_dce110.c |   2 +-
 .../diagnostics/hw_ctx_adapter_service_diag.c      |   2 +-
 .../amd/dal/dc/adapter/hw_ctx_adapter_service.c    |   2 +-
 .../drm/amd/dal/dc/adapter/wireless_data_source.c  |   2 +-
 .../amd/dal/dc/asic_capability/asic_capability.c   |   2 +-
 .../dc/asic_capability/carrizo_asic_capability.c   |   2 +-
 .../dal/dc/asic_capability/tonga_asic_capability.c |   2 +-
 drivers/gpu/drm/amd/dal/dc/audio/audio_base.c      |   2 +-
 .../gpu/drm/amd/dal/dc/audio/dce110/audio_dce110.c |   2 +-
 .../amd/dal/dc/audio/dce110/hw_ctx_audio_dce110.c  |   2 +-
 drivers/gpu/drm/amd/dal/dc/audio/hw_ctx_audio.c    |   2 +-
 drivers/gpu/drm/amd/dal/dc/basics/conversion.c     |   2 +-
 drivers/gpu/drm/amd/dal/dc/basics/fixpt31_32.c     |   2 +-
 drivers/gpu/drm/amd/dal/dc/basics/fixpt32_32.c     |   2 +-
 drivers/gpu/drm/amd/dal/dc/basics/grph_object_id.c |   2 +-
 drivers/gpu/drm/amd/dal/dc/basics/logger.c         |   2 +-
 .../gpu/drm/amd/dal/dc/basics/register_logger.c    |   2 +-
 drivers/gpu/drm/amd/dal/dc/basics/vector.c         |   2 +-
 drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c      |   2 +-
 .../gpu/drm/amd/dal/dc/bios/bios_parser_helper.c   |   2 +-
 drivers/gpu/drm/amd/dal/dc/bios/command_table.c    |   2 +-
 .../gpu/drm/amd/dal/dc/bios/command_table_helper.c |   2 +-
 .../dal/dc/bios/dce110/bios_parser_helper_dce110.c |   2 +-
 .../dc/bios/dce110/command_table_helper_dce110.c   |   2 +-
 drivers/gpu/drm/amd/dal/dc/calcs/scaler_filter.c   |   2 +-
 drivers/gpu/drm/amd/dal/dc/dc_services.h           | 227 ++++++++++++++++++++-
 drivers/gpu/drm/amd/dal/dc/dc_services_types.h     | 167 +++++++++++++++
 drivers/gpu/drm/amd/dal/dc/dc_types.h              |   1 -
 .../drm/amd/dal/dc/dce100/dce100_hw_sequencer.c    |   2 +
 .../gpu/drm/amd/dal/dc/dce110/dce110_compressor.c  |   2 +-
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp.c     |   2 +-
 .../gpu/drm/amd/dal/dc/dce110/dce110_ipp_cursor.c  |   2 +-
 .../gpu/drm/amd/dal/dc/dce110/dce110_ipp_gamma.c   |   2 +-
 .../drm/amd/dal/dc/dce110/dce110_link_encoder.c    |   2 +-
 .../gpu/drm/amd/dal/dc/dce110/dce110_mem_input.c   |   2 +-
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp.c     |   2 +-
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_csc.c |   2 +-
 .../drm/amd/dal/dc/dce110/dce110_opp_formatter.c   |   2 +-
 .../gpu/drm/amd/dal/dc/dce110/dce110_opp_regamma.c |   2 +-
 .../drm/amd/dal/dc/dce110/dce110_stream_encoder.c  |   2 +-
 .../amd/dal/dc/dce110/dce110_timing_generator.c    |   2 +-
 .../gpu/drm/amd/dal/dc/dce110/dce110_transform.c   |   2 +-
 .../amd/dal/dc/dce110/dce110_transform_bit_depth.c |   2 +-
 .../drm/amd/dal/dc/dce110/dce110_transform_gamut.c |   2 +-
 .../drm/amd/dal/dc/dce110/dce110_transform_scl.c   |   2 +-
 .../drm/amd/dal/dc/dce110/dce110_transform_sclv.c  |   2 +-
 drivers/gpu/drm/amd/dal/dc/dcs/ddc_i2caux_helper.c |   2 +-
 drivers/gpu/drm/amd/dal/dc/dcs/ddc_service.c       |   4 +-
 .../gpu/drm/amd/dal/dc/gpio/dce110/hw_ddc_dce110.c |   2 +-
 .../drm/amd/dal/dc/gpio/dce110/hw_factory_dce110.c |   2 +-
 .../gpu/drm/amd/dal/dc/gpio/dce110/hw_hpd_dce110.c |   2 +-
 .../amd/dal/dc/gpio/dce110/hw_translate_dce110.c   |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/ddc.c              |   2 +-
 .../drm/amd/dal/dc/gpio/diagnostics/hw_ddc_diag.c  |   2 +-
 .../amd/dal/dc/gpio/diagnostics/hw_factory_diag.c  |   2 +-
 .../drm/amd/dal/dc/gpio/diagnostics/hw_hpd_diag.c  |   2 +-
 .../dal/dc/gpio/diagnostics/hw_translate_diag.c    |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/gpio_base.c        |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/gpio_service.c     |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/hw_ddc.c           |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/hw_factory.c       |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio.c          |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio_pad.c      |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio_pin.c      |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/hw_hpd.c           |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/hw_translate.c     |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpio/irq.c              |   2 +-
 .../gpu/drm/amd/dal/dc/gpu/calc_pll_clock_source.c |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpu/clock_source.c      |   2 +-
 .../gpu/drm/amd/dal/dc/gpu/dc_clock_generator.c    |   2 +-
 .../amd/dal/dc/gpu/dce110/dc_clock_gating_dce110.c |   2 +-
 .../amd/dal/dc/gpu/dce110/display_clock_dce110.c   |   2 +-
 .../dal/dc/gpu/dce110/ext_clock_source_dce110.c    |   2 +-
 .../dal/dc/gpu/dce110/pll_clock_source_dce110.c    |   2 +-
 .../dal/dc/gpu/dce110/vce_clock_source_dce110.c    |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpu/display_clock.c     |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpu/divider_range.c     |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpu/ext_clock_source.c  |   2 +-
 drivers/gpu/drm/amd/dal/dc/gpu/pll_clock_source.c  |   2 +-
 drivers/gpu/drm/amd/dal/dc/i2caux/aux_engine.c     |   2 +-
 .../amd/dal/dc/i2caux/dce110/aux_engine_dce110.c   |   2 +-
 .../dal/dc/i2caux/dce110/i2c_hw_engine_dce110.c    |   2 +-
 .../dal/dc/i2caux/dce110/i2c_sw_engine_dce110.c    |   2 +-
 .../drm/amd/dal/dc/i2caux/dce110/i2caux_dce110.c   |   2 +-
 .../amd/dal/dc/i2caux/diagnostics/i2caux_diag.c    |   2 +-
 drivers/gpu/drm/amd/dal/dc/i2caux/engine_base.c    |   2 +-
 drivers/gpu/drm/amd/dal/dc/i2caux/i2c_engine.c     |   2 +-
 .../drm/amd/dal/dc/i2caux/i2c_generic_hw_engine.c  |   2 +-
 drivers/gpu/drm/amd/dal/dc/i2caux/i2c_hw_engine.c  |   2 +-
 drivers/gpu/drm/amd/dal/dc/i2caux/i2c_sw_engine.c  |   2 +-
 drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c         |   2 +-
 drivers/gpu/drm/amd/dal/dc/inc/core_dc.h           |   1 -
 drivers/gpu/drm/amd/dal/dc/inc/link_encoder.h      |   1 -
 .../drm/amd/dal/dc/irq/dce110/irq_service_dce110.c |   2 +-
 drivers/gpu/drm/amd/dal/dc/irq/irq_service.c       |   2 +-
 drivers/gpu/drm/amd/dal/dc/irq_types.h             |   2 -
 .../drm/amd/dal/dc/virtual/virtual_link_encoder.c  |   2 +
 .../amd/dal/dc/virtual/virtual_stream_encoder.c    |   3 +-
 drivers/gpu/drm/amd/dal/include/dal_types.h        |   1 -
 drivers/gpu/drm/amd/dal/include/fixed31_32.h       |   2 -
 drivers/gpu/drm/amd/dal/include/fixed32_32.h       |   3 -
 drivers/gpu/drm/amd/dal/include/grph_object_id.h   |   2 -
 .../gpu/drm/amd/dal/include/link_service_types.h   |   2 -
 drivers/gpu/drm/amd/dal/include/signal_types.h     |   2 -
 111 files changed, 492 insertions(+), 194 deletions(-)
 delete mode 100644 drivers/gpu/drm/amd/dal/dal_power_interface_types.h
 create mode 100644 drivers/gpu/drm/amd/dal/dc/dc_services_types.h

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dal_services.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dal_services.c
index b67599d..eec5313 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dal_services.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dal_services.c
@@ -30,7 +30,7 @@
 #include <drm/drm_crtc_helper.h>
 #include <drm/amdgpu_drm.h>
 #include "amdgpu.h"
-#include "dal_services.h"
+#include "dc_services.h"
 #include "amdgpu_dm.h"
 #include "amdgpu_dm_irq.h"
 #include "amdgpu_dm_types.h"
diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
index 2cece0f..bb65892 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services_types.h"
+#include "dc_services_types.h"
 #include "dc.h"
 
 #include "vid.h"
diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_irq.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_irq.c
index 7cfb754..b624229 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_irq.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_irq.c
@@ -25,7 +25,7 @@
 
 #include <drm/drmP.h>
 
-#include "dal_services_types.h"
+#include "dc_services_types.h"
 #include "dc.h"
 
 #include "amdgpu.h"
diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
index fe5e366..f52b2f2 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
@@ -24,7 +24,7 @@
  */
 
 #include <drm/drm_atomic_helper.h>
-#include "dal_services.h"
+#include "dc_services.h"
 #include "amdgpu.h"
 #include "amdgpu_dm_types.h"
 #include "amdgpu_dm_mst_types.h"
diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
index c472530..5fe5ca4 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services_types.h"
+#include "dc_services_types.h"
 
 #include <linux/types.h>
 #include <drm/drmP.h>
diff --git a/drivers/gpu/drm/amd/dal/dal_power_interface_types.h b/drivers/gpu/drm/amd/dal/dal_power_interface_types.h
deleted file mode 100644
index 82e8ca2..0000000
--- a/drivers/gpu/drm/amd/dal/dal_power_interface_types.h
+++ /dev/null
@@ -1,76 +0,0 @@
-/*
- * Copyright 2015 Advanced Micro Devices, Inc.
- *
- * Permission is hereby granted, free of charge, to any person obtaining a
- * copy of this software and associated documentation files (the "Software"),
- * to deal in the Software without restriction, including without limitation
- * the rights to use, copy, modify, merge, publish, distribute, sublicense,
- * and/or sell copies of the Software, and to permit persons to whom the
- * Software is furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in
- * all copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
- * THE COPYRIGHT HOLDER(S) OR AUTHOR(S) BE LIABLE FOR ANY CLAIM, DAMAGES OR
- * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
- * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- * OTHER DEALINGS IN THE SOFTWARE.
- *
- * Authors: AMD
- *
- */
-
-#ifndef __DAL_POWER_INTERFACE_TYPES_H__
-#define __DAL_POWER_INTERFACE_TYPES_H__
-
-enum dal_to_power_clocks_state {
-	PP_CLOCKS_STATE_INVALID,
-	PP_CLOCKS_STATE_ULTRA_LOW,
-	PP_CLOCKS_STATE_LOW,
-	PP_CLOCKS_STATE_NOMINAL,
-	PP_CLOCKS_STATE_PERFORMANCE
-};
-
-/* clocks in khz */
-struct dal_to_power_info {
-	enum dal_to_power_clocks_state required_clock;
-	uint32_t min_sclk;
-	uint32_t min_mclk;
-	uint32_t min_deep_sleep_sclk;
-};
-
-/* clocks in khz */
-struct power_to_dal_info {
-	uint32_t min_sclk;
-	uint32_t max_sclk;
-	uint32_t min_mclk;
-	uint32_t max_mclk;
-};
-
-/* clocks in khz */
-struct dal_system_clock_range {
-	uint32_t min_sclk;
-	uint32_t max_sclk;
-
-	uint32_t min_mclk;
-	uint32_t max_mclk;
-
-	uint32_t min_dclk;
-	uint32_t max_dclk;
-
-	/* Wireless Display */
-	uint32_t min_eclk;
-	uint32_t max_eclk;
-};
-
-/* clocks in khz */
-struct dal_to_power_dclk {
-	uint32_t optimal; /* input: best optimizes for stutter efficiency */
-	uint32_t minimal; /* input: the lowest clk that DAL can support */
-	uint32_t established; /* output: the actually set one */
-};
-
-#endif /* __DAL_POWER_INTERFACE_TYPES_H__ */
diff --git a/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c b/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
index d0dd6c9..9fb1be8 100644
--- a/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
+++ b/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
@@ -24,7 +24,7 @@
  */
 
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "dc_bios_types.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/adapter/dce110/hw_ctx_adapter_service_dce110.c b/drivers/gpu/drm/amd/dal/dc/adapter/dce110/hw_ctx_adapter_service_dce110.c
index 31c2aab..98b1475 100644
--- a/drivers/gpu/drm/amd/dal/dc/adapter/dce110/hw_ctx_adapter_service_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/adapter/dce110/hw_ctx_adapter_service_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "../hw_ctx_adapter_service.h"
 
 #include "hw_ctx_adapter_service_dce110.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/adapter/diagnostics/hw_ctx_adapter_service_diag.c b/drivers/gpu/drm/amd/dal/dc/adapter/diagnostics/hw_ctx_adapter_service_diag.c
index de45ce3..ba377f4 100644
--- a/drivers/gpu/drm/amd/dal/dc/adapter/diagnostics/hw_ctx_adapter_service_diag.c
+++ b/drivers/gpu/drm/amd/dal/dc/adapter/diagnostics/hw_ctx_adapter_service_diag.c
@@ -25,7 +25,7 @@
 
 /* FPGA Diagnostics version of AS HW CTX. */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "../hw_ctx_adapter_service.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/adapter/hw_ctx_adapter_service.c b/drivers/gpu/drm/amd/dal/dc/adapter/hw_ctx_adapter_service.c
index 5fa886f..0d13a90 100644
--- a/drivers/gpu/drm/amd/dal/dc/adapter/hw_ctx_adapter_service.c
+++ b/drivers/gpu/drm/amd/dal/dc/adapter/hw_ctx_adapter_service.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/adapter_service_types.h"
 #include "include/grph_object_id.h"
 #include "hw_ctx_adapter_service.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/adapter/wireless_data_source.c b/drivers/gpu/drm/amd/dal/dc/adapter/wireless_data_source.c
index 0249829..122222e 100644
--- a/drivers/gpu/drm/amd/dal/dc/adapter/wireless_data_source.c
+++ b/drivers/gpu/drm/amd/dal/dc/adapter/wireless_data_source.c
@@ -24,7 +24,7 @@
  */
 
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "adapter_service.h"
 #include "wireless_data_source.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/asic_capability/asic_capability.c b/drivers/gpu/drm/amd/dal/dc/asic_capability/asic_capability.c
index b3eb665..05a92a4 100644
--- a/drivers/gpu/drm/amd/dal/dc/asic_capability/asic_capability.c
+++ b/drivers/gpu/drm/amd/dal/dc/asic_capability/asic_capability.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/logger_interface.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/asic_capability/carrizo_asic_capability.c b/drivers/gpu/drm/amd/dal/dc/asic_capability/carrizo_asic_capability.c
index b106ccc..1de790d 100644
--- a/drivers/gpu/drm/amd/dal/dc/asic_capability/carrizo_asic_capability.c
+++ b/drivers/gpu/drm/amd/dal/dc/asic_capability/carrizo_asic_capability.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/asic_capability_interface.h"
 #include "include/asic_capability_types.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/asic_capability/tonga_asic_capability.c b/drivers/gpu/drm/amd/dal/dc/asic_capability/tonga_asic_capability.c
index 599c47d..7cd0b80 100644
--- a/drivers/gpu/drm/amd/dal/dc/asic_capability/tonga_asic_capability.c
+++ b/drivers/gpu/drm/amd/dal/dc/asic_capability/tonga_asic_capability.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/asic_capability_interface.h"
 #include "include/asic_capability_types.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/audio/audio_base.c b/drivers/gpu/drm/amd/dal/dc/audio/audio_base.c
index 2737851..2311f29 100644
--- a/drivers/gpu/drm/amd/dal/dc/audio/audio_base.c
+++ b/drivers/gpu/drm/amd/dal/dc/audio/audio_base.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/logger_interface.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/audio/dce110/audio_dce110.c b/drivers/gpu/drm/amd/dal/dc/audio/dce110/audio_dce110.c
index f284870..5927b12 100644
--- a/drivers/gpu/drm/amd/dal/dc/audio/dce110/audio_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/audio/dce110/audio_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/logger_interface.h"
 
 #include "audio_dce110.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/audio/dce110/hw_ctx_audio_dce110.c b/drivers/gpu/drm/amd/dal/dc/audio/dce110/hw_ctx_audio_dce110.c
index a13b2ab..288f14f 100644
--- a/drivers/gpu/drm/amd/dal/dc/audio/dce110/hw_ctx_audio_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/audio/dce110/hw_ctx_audio_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/logger_interface.h"
 #include "../hw_ctx_audio.h"
 #include "hw_ctx_audio_dce110.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/audio/hw_ctx_audio.c b/drivers/gpu/drm/amd/dal/dc/audio/hw_ctx_audio.c
index f1f1298..a78ab79 100644
--- a/drivers/gpu/drm/amd/dal/dc/audio/hw_ctx_audio.c
+++ b/drivers/gpu/drm/amd/dal/dc/audio/hw_ctx_audio.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "hw_ctx_audio.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/basics/conversion.c b/drivers/gpu/drm/amd/dal/dc/basics/conversion.c
index 8c38206..0eb7813 100644
--- a/drivers/gpu/drm/amd/dal/dc/basics/conversion.c
+++ b/drivers/gpu/drm/amd/dal/dc/basics/conversion.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #define DIVIDER 10000
 
diff --git a/drivers/gpu/drm/amd/dal/dc/basics/fixpt31_32.c b/drivers/gpu/drm/amd/dal/dc/basics/fixpt31_32.c
index 6ce75b3..68626ba 100644
--- a/drivers/gpu/drm/amd/dal/dc/basics/fixpt31_32.c
+++ b/drivers/gpu/drm/amd/dal/dc/basics/fixpt31_32.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/fixed31_32.h"
 
 static inline uint64_t abs_i64(
diff --git a/drivers/gpu/drm/amd/dal/dc/basics/fixpt32_32.c b/drivers/gpu/drm/amd/dal/dc/basics/fixpt32_32.c
index 1140132..c52fe47 100644
--- a/drivers/gpu/drm/amd/dal/dc/basics/fixpt32_32.c
+++ b/drivers/gpu/drm/amd/dal/dc/basics/fixpt32_32.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/fixed32_32.h"
 
 static uint64_t u64_div(uint64_t n, uint64_t d)
diff --git a/drivers/gpu/drm/amd/dal/dc/basics/grph_object_id.c b/drivers/gpu/drm/amd/dal/dc/basics/grph_object_id.c
index 8276f9d..714a571 100644
--- a/drivers/gpu/drm/amd/dal/dc/basics/grph_object_id.c
+++ b/drivers/gpu/drm/amd/dal/dc/basics/grph_object_id.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/grph_object_id.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/basics/logger.c b/drivers/gpu/drm/amd/dal/dc/basics/logger.c
index 9c2a889..49cef8a 100644
--- a/drivers/gpu/drm/amd/dal/dc/basics/logger.c
+++ b/drivers/gpu/drm/amd/dal/dc/basics/logger.c
@@ -23,7 +23,7 @@
  *
  */
 #include <stdarg.h>
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/dal_types.h"
 #include "include/logger_interface.h"
 #include "logger.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/basics/register_logger.c b/drivers/gpu/drm/amd/dal/dc/basics/register_logger.c
index a3086a0..5dcf3fc 100644
--- a/drivers/gpu/drm/amd/dal/dc/basics/register_logger.c
+++ b/drivers/gpu/drm/amd/dal/dc/basics/register_logger.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/dal_types.h"
 #include "include/logger_interface.h"
 #include "logger.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/basics/vector.c b/drivers/gpu/drm/amd/dal/dc/basics/vector.c
index 2f932c0..ea682a7 100644
--- a/drivers/gpu/drm/amd/dal/dc/basics/vector.c
+++ b/drivers/gpu/drm/amd/dal/dc/basics/vector.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/vector.h"
 
 bool dal_vector_construct(
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
index 8d675f0..2ef2543 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "atom.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.c b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.c
index da559b0..fe05df2 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "atom.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/command_table.c b/drivers/gpu/drm/amd/dal/dc/bios/command_table.c
index cba54f3..1a27bc8 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/command_table.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/command_table.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "atom.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/command_table_helper.c b/drivers/gpu/drm/amd/dal/dc/bios/command_table_helper.c
index 36d1240..d379496 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/command_table_helper.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/command_table_helper.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "atom.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/dce110/bios_parser_helper_dce110.c b/drivers/gpu/drm/amd/dal/dc/bios/dce110/bios_parser_helper_dce110.c
index 2cc2d2d..f6c7df5 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/dce110/bios_parser_helper_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/dce110/bios_parser_helper_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "atom.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/dce110/command_table_helper_dce110.c b/drivers/gpu/drm/amd/dal/dc/bios/dce110/command_table_helper_dce110.c
index 0319382..182029b 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/dce110/command_table_helper_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/dce110/command_table_helper_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "atom.h"
 
 #include "include/bios_parser_types.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/calcs/scaler_filter.c b/drivers/gpu/drm/amd/dal/dc/calcs/scaler_filter.c
index f8ee65e..3dd8781 100644
--- a/drivers/gpu/drm/amd/dal/dc/calcs/scaler_filter.c
+++ b/drivers/gpu/drm/amd/dal/dc/calcs/scaler_filter.c
@@ -22,7 +22,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/fixed31_32.h"
 
 #include "scaler_filter.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dc_services.h b/drivers/gpu/drm/amd/dal/dc/dc_services.h
index b8b8b20..fa1c39d 100644
--- a/drivers/gpu/drm/amd/dal/dc/dc_services.h
+++ b/drivers/gpu/drm/amd/dal/dc/dc_services.h
@@ -31,16 +31,20 @@
 #define __DC_SERVICES_H__
 
 /* TODO: remove when DC is complete. */
-#include "dal_services_types.h"
+#include "dc_services_types.h"
 #include "logger_interface.h"
 #include "include/dal_types.h"
 #include "irq_types.h"
-#include "dal_power_interface_types.h"
 #include "link_service_types.h"
 
+#undef DEPRECATED
+
 /* if the pointer is not NULL, the allocated memory is zeroed */
 void *dc_service_alloc(struct dc_context *ctx, uint32_t size);
 
+/* reallocate memory. The contents will remain unchanged.*/
+void *dc_service_realloc(struct dc_context *ctx, const void *ptr, uint32_t size);
+
 void dc_service_free(struct dc_context *ctx, void *p);
 
 void dc_service_memset(void *p, int32_t c, uint32_t count);
@@ -50,6 +54,8 @@ void dc_service_memmove(void *dst, const void *src, uint32_t size);
 /* TODO: rename to dc_memcmp*/
 int32_t dal_memcmp(const void *p1, const void *p2, uint32_t count);
 
+int32_t dal_strncmp(const int8_t *p1, const int8_t *p2, uint32_t count);
+
 irq_handler_idx dc_service_register_interrupt(
 	struct dc_context *ctx,
 	struct dc_interrupt_params *int_params,
@@ -61,10 +67,182 @@ void dc_service_unregister_interrupt(
 	enum dc_irq_source irq_source,
 	irq_handler_idx handler_idx);
 
+/*
+ *
+ * GPU registers access
+ *
+ */
+static inline uint32_t dal_read_reg(
+	const struct dc_context *ctx,
+	uint32_t address)
+{
+	uint32_t value = cgs_read_register(ctx->cgs_device, address);
+
+#if defined(__DAL_REGISTER_LOGGER__)
+	if (true == dal_reg_logger_should_dump_register()) {
+		dal_reg_logger_rw_count_increment();
+		DRM_INFO("%s 0x%x 0x%x\n", __func__, address, value);
+	}
+#endif
+	return value;
+}
+
+static inline void dal_write_reg(
+	const struct dc_context *ctx,
+	uint32_t address,
+	uint32_t value)
+{
+#if defined(__DAL_REGISTER_LOGGER__)
+	if (true == dal_reg_logger_should_dump_register()) {
+		dal_reg_logger_rw_count_increment();
+		DRM_INFO("%s 0x%x 0x%x\n", __func__, address, value);
+	}
+#endif
+	cgs_write_register(ctx->cgs_device, address, value);
+}
+
+static inline uint32_t dal_read_index_reg(
+	const struct dc_context *ctx,
+	enum cgs_ind_reg addr_space,
+	uint32_t index)
+{
+	return cgs_read_ind_register(ctx->cgs_device, addr_space, index);
+}
+
+static inline void dal_write_index_reg(
+	const struct dc_context *ctx,
+	enum cgs_ind_reg addr_space,
+	uint32_t index,
+	uint32_t value)
+{
+	cgs_write_ind_register(ctx->cgs_device, addr_space, index, value);
+}
+
+static inline uint32_t get_reg_field_value_ex(
+	uint32_t reg_value,
+	uint32_t mask,
+	uint8_t shift)
+{
+	return (mask & reg_value) >> shift;
+}
+
+#define get_reg_field_value(reg_value, reg_name, reg_field)\
+	get_reg_field_value_ex(\
+		(reg_value),\
+		reg_name ## __ ## reg_field ## _MASK,\
+		reg_name ## __ ## reg_field ## __SHIFT)
+
+static inline uint32_t set_reg_field_value_ex(
+	uint32_t reg_value,
+	uint32_t value,
+	uint32_t mask,
+	uint8_t shift)
+{
+	return (reg_value & ~mask) | (mask & (value << shift));
+}
+
+#define set_reg_field_value(reg_value, value, reg_name, reg_field)\
+	(reg_value) = set_reg_field_value_ex(\
+		(reg_value),\
+		(value),\
+		reg_name ## __ ## reg_field ## _MASK,\
+		reg_name ## __ ## reg_field ## __SHIFT)
+
+/*
+ * atombios services
+ */
+
+bool dal_exec_bios_cmd_table(
+	struct dc_context *ctx,
+	uint32_t index,
+	void *params);
+
+#ifdef BUILD_DAL_TEST
+uint32_t dal_bios_cmd_table_para_revision(
+struct dc_context *ctx,
+	uint32_t index);
+
+bool dal_bios_cmd_table_revision(
+	struct dc_context *ctx,
+	uint32_t index,
+	uint8_t *frev,
+	uint8_t *crev);
+#endif
+
+#ifndef BUILD_DAL_TEST
+static inline uint32_t dal_bios_cmd_table_para_revision(
+	struct dc_context *ctx,
+	uint32_t index)
+{
+	uint8_t frev;
+	uint8_t crev;
+
+	if (cgs_atom_get_cmd_table_revs(
+			ctx->cgs_device,
+			index,
+			&frev,
+			&crev) != 0)
+		return 0;
+
+	return crev;
+}
+#else
+uint32_t dal_bios_cmd_table_para_revision(
+		struct dc_context *ctx,
+		uint32_t index);
+#endif
+
 /**************************************
  * Power Play (PP) interfaces
  **************************************/
 
+enum dal_to_power_clocks_state {
+	PP_CLOCKS_STATE_INVALID,
+	PP_CLOCKS_STATE_ULTRA_LOW,
+	PP_CLOCKS_STATE_LOW,
+	PP_CLOCKS_STATE_NOMINAL,
+	PP_CLOCKS_STATE_PERFORMANCE
+};
+
+/* clocks in khz */
+struct dal_to_power_info {
+	enum dal_to_power_clocks_state required_clock;
+	uint32_t min_sclk;
+	uint32_t min_mclk;
+	uint32_t min_deep_sleep_sclk;
+};
+
+/* clocks in khz */
+struct power_to_dal_info {
+	uint32_t min_sclk;
+	uint32_t max_sclk;
+	uint32_t min_mclk;
+	uint32_t max_mclk;
+};
+
+/* clocks in khz */
+struct dal_system_clock_range {
+	uint32_t min_sclk;
+	uint32_t max_sclk;
+
+	uint32_t min_mclk;
+	uint32_t max_mclk;
+
+	uint32_t min_dclk;
+	uint32_t max_dclk;
+
+	/* Wireless Display */
+	uint32_t min_eclk;
+	uint32_t max_eclk;
+};
+
+/* clocks in khz */
+struct dal_to_power_dclk {
+	uint32_t optimal; /* input: best optimizes for stutter efficiency */
+	uint32_t minimal; /* input: the lowest clk that DAL can support */
+	uint32_t established; /* output: the actually set one */
+};
+
 /* DAL calls this function to notify PP about clocks it needs for the Mode Set.
  * This is done *before* it changes DCE clock.
  *
@@ -224,10 +402,51 @@ bool dc_service_pp_apply_display_requirements(
 
 void dc_service_sleep_in_milliseconds(struct dc_context *ctx, uint32_t milliseconds);
 
-/* end of power component calls */
-
 void dc_service_delay_in_microseconds(struct dc_context *ctx, uint32_t microseconds);
 
+enum platform_method {
+	PM_GET_AVAILABLE_METHODS = 1 << 0,
+	PM_GET_LID_STATE = 1 << 1,
+	PM_GET_EXTENDED_BRIGHNESS_CAPS = 1 << 2
+};
+
+struct platform_info_params {
+	enum platform_method method;
+	void *data;
+};
+
+struct platform_info_brightness_caps {
+	uint8_t ac_level_percentage;
+	uint8_t dc_level_percentage;
+};
+
+struct platform_info_ext_brightness_caps {
+	struct platform_info_brightness_caps basic_caps;
+	struct data_point {
+		uint8_t luminance;
+		uint8_t	signal_level;
+	} data_points[99];
+
+	uint8_t	data_points_num;
+	uint8_t	min_input_signal;
+	uint8_t	max_input_signal;
+};
+
+bool dal_get_platform_info(
+	struct dc_context *ctx,
+	struct platform_info_params *params);
+
+/*
+ *
+ * print-out services
+ *
+ */
+#define dal_log_to_buffer(buffer, size, fmt, args)\
+	vsnprintf(buffer, size, fmt, args)
+
+long dal_get_pid(void);
+long dal_get_tgid(void);
+
 /*
  *
  * general debug capabilities
diff --git a/drivers/gpu/drm/amd/dal/dc/dc_services_types.h b/drivers/gpu/drm/amd/dal/dc/dc_services_types.h
new file mode 100644
index 0000000..aded7b1
--- /dev/null
+++ b/drivers/gpu/drm/amd/dal/dc/dc_services_types.h
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2012-15 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER(S) OR AUTHOR(S) BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Authors: AMD
+ *
+ */
+
+#ifndef __DC_SERVICES_TYPES_H__
+#define __DC_SERVICES_TYPES_H__
+
+#define INVALID_DISPLAY_INDEX 0xffffffff
+
+#if defined __KERNEL__
+
+#include <asm/byteorder.h>
+#include <linux/types.h>
+#include <drm/drmP.h>
+
+#include "cgs_linux.h"
+
+#if defined(__BIG_ENDIAN) && !defined(BIGENDIAN_CPU)
+#define BIGENDIAN_CPU
+#elif defined(__LITTLE_ENDIAN) && !defined(LITTLEENDIAN_CPU)
+#define LITTLEENDIAN_CPU
+#endif
+
+#undef READ
+#undef WRITE
+#undef FRAME_SIZE
+
+#define dal_output_to_console(fmt, ...) DRM_INFO(fmt, ##__VA_ARGS__)
+
+#define dal_error(fmt, ...) DRM_ERROR(fmt, ##__VA_ARGS__)
+
+#define dal_debug(fmt, ...) DRM_DEBUG_KMS(fmt, ##__VA_ARGS__)
+
+#define dal_vlog(fmt, args) vprintk(fmt, args)
+
+#define dal_min(x, y) min(x, y)
+#define dal_max(x, y) max(x, y)
+
+#elif defined BUILD_DAL_TEST
+
+#include <inttypes.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <stdio.h>
+
+#include <stdarg.h>
+
+#include "cgs_linux.h"
+
+#define LONG_MAX	((long)(~0UL>>1))
+#define LONG_MIN	(-LONG_MAX - 1)
+#define LLONG_MAX	((long long)(~0ULL>>1))
+#define LLONG_MIN	(-LLONG_MAX - 1)
+#define UINT_MAX	(~0U)
+
+typedef _Bool bool;
+enum { false, true };
+
+#ifndef NULL
+#define NULL ((void *)0)
+#endif
+
+#define LITTLEENDIAN_CPU 1
+
+#include <test_context.h>
+
+#define offsetof(TYPE, MEMBER) ((size_t) &((TYPE *)0)->MEMBER)
+
+#define container_of(ptr, type, member) \
+	((type *)((char *)(ptr) - offsetof(type, member)))
+
+#define dal_test_not_implemented() \
+	printf("[DAL_TEST_NOT_IMPL]:%s\n", __func__)
+
+#define dal_output_to_console(fmt, ...) do { \
+	printf("[DAL_LOG]" fmt, ##__VA_ARGS__); } \
+	while (false)
+
+#define dal_error(fmt, ...) printf("[DAL_ERROR]" fmt, ##__VA_ARGS__)
+
+#define dal_output_to_console(fmt, ...) do { \
+			printf("[DAL_LOG]" fmt, ##__VA_ARGS__); } \
+				while (false)
+
+
+#define dal_debug(fmt, ...) printf("[DAL_DBG]" fmt, ##__VA_ARGS__)
+
+#define dal_vlog(fmt, args) vprintf(fmt, args)
+
+#define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))
+
+#define dal_min(x, y) ({\
+	typeof(x) _min1 = (x);\
+	typeof(y) _min2 = (y);\
+	(void) (&_min1 == &_min2);\
+	_min1 < _min2 ? _min1 : _min2; })
+
+#define dal_max(x, y) ({\
+	typeof(x) _max1 = (x);\
+	typeof(y) _max2 = (y);\
+	(void) (&_max1 == &_max2);\
+	_max1 > _max2 ? _max1 : _max2; })
+
+/* division functions */
+
+static inline int64_t div64_s64(int64_t x, int64_t y)
+{
+	return x / y;
+}
+
+static inline uint64_t div64_u64(uint64_t x, uint64_t y)
+{
+	return x / y;
+}
+
+static inline uint64_t div_u64(uint64_t x, uint32_t y)
+{
+	return x / y;
+}
+
+static inline uint64_t div64_u64_rem(uint64_t x, uint64_t y, uint64_t *rem)
+{
+	if (rem)
+		*rem = x % y;
+	return x / y;
+}
+
+static inline uint64_t div_u64_rem(uint64_t x, uint32_t y, uint32_t *rem)
+{
+	if (rem)
+		*rem = x % y;
+	return x / y;
+}
+
+#define cpu_to_le16(do_nothing) do_nothing
+
+#define le16_to_cpu(do_nothing) do_nothing
+
+#define cpu_to_le32(do_nothing) do_nothing
+
+#define le32_to_cpu(do_nothing) do_nothing
+
+#endif
+
+#endif
diff --git a/drivers/gpu/drm/amd/dal/dc/dc_types.h b/drivers/gpu/drm/amd/dal/dc/dc_types.h
index 18ecb0d..c797642 100644
--- a/drivers/gpu/drm/amd/dal/dc/dc_types.h
+++ b/drivers/gpu/drm/amd/dal/dc/dc_types.h
@@ -25,7 +25,6 @@
 #ifndef DC_TYPES_H_
 #define DC_TYPES_H_
 
-#include "dal_services_types.h"
 #include "fixed32_32.h"
 #include "fixed31_32.h"
 #include "irq_types.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce100/dce100_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce100/dce100_hw_sequencer.c
index a235bb0..b37df4a 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce100/dce100_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce100/dce100_hw_sequencer.c
@@ -23,6 +23,8 @@
  *
  */
 
+#include "dc_services.h"
+
 #include "dce110/dce110_hw_sequencer.h"
 
 #include "resource.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_compressor.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_compressor.c
index 7abb790..6761b4f 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_compressor.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_compressor.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "dce/dce_11_0_d.h"
 #include "dce/dce_11_0_sh_mask.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp.c
index a29dc51..6cd80ae 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/logger_interface.h"
 
 #include "dce/dce_11_0_d.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp_cursor.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp_cursor.c
index 08b7940..0569fbb 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp_cursor.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp_cursor.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/logger_interface.h"
 
 #include "dce/dce_11_0_d.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp_gamma.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp_gamma.c
index b7186b1..2aa432a 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp_gamma.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp_gamma.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/logger_interface.h"
 #include "include/fixed31_32.h"
 #include "basics/conversion.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_link_encoder.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_link_encoder.c
index 4a83e25..2396f15 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_link_encoder.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_link_encoder.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "core_types.h"
 #include "link_encoder.h"
 #include "stream_encoder.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_mem_input.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_mem_input.c
index f5cf5ad..4e809b6 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_mem_input.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_mem_input.c
@@ -22,7 +22,7 @@
  * Authors: AMD
  *
  */
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "dce/dce_11_0_d.h"
 #include "dce/dce_11_0_sh_mask.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp.c
index 5003c89..99163ee 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /* include DCE11 register header files */
 #include "dce/dce_11_0_d.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_csc.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_csc.c
index 91430c0..a96a72a 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_csc.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_csc.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "dce110_opp.h"
 #include "basics/conversion.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_formatter.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_formatter.c
index fdf87bd..0224ade 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_formatter.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_formatter.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "dce/dce_11_0_d.h"
 #include "dce/dce_11_0_sh_mask.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_regamma.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_regamma.c
index cf116f1..dceba7d 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_regamma.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_regamma.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /* include DCE11 register header files */
 #include "dce/dce_11_0_d.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.c
index 5b432d8..fadcc06 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "dc_bios_types.h"
 #include "dce110_stream_encoder.h"
 #include "dce/dce_11_0_d.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_timing_generator.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_timing_generator.c
index 1bb89d8..79e34dc 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_timing_generator.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_timing_generator.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /* include DCE11 register header files */
 #include "dce/dce_11_0_d.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform.c
index 5a87ded..16cddb5 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /* include DCE11 register header files */
 #include "dce/dce_11_0_d.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_bit_depth.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_bit_depth.c
index 3e0d151..d0e4b98 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_bit_depth.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_bit_depth.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /* include DCE11 register header files */
 #include "dce/dce_11_0_d.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_gamut.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_gamut.c
index 0dd4355..bb3b3cc 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_gamut.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_gamut.c
@@ -22,7 +22,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "dce110_transform.h"
 #include "dce/dce_11_0_d.h"
 #include "dce/dce_11_0_sh_mask.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_scl.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_scl.c
index f313d2c..4ba14c2 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_scl.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_scl.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /* include DCE11 register header files */
 #include "dce/dce_11_0_d.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_sclv.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_sclv.c
index bcf20bb..9b25ed7 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_sclv.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_transform_sclv.c
@@ -22,7 +22,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "dce/dce_11_0_d.h"
 #include "dce/dce_11_0_sh_mask.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dcs/ddc_i2caux_helper.c b/drivers/gpu/drm/amd/dal/dc/dcs/ddc_i2caux_helper.c
index a4442d6..0af4df6 100644
--- a/drivers/gpu/drm/amd/dal/dc/dcs/ddc_i2caux_helper.c
+++ b/drivers/gpu/drm/amd/dal/dc/dcs/ddc_i2caux_helper.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "ddc_i2caux_helper.h"
 #include "include/ddc_service_types.h"
 #include "include/vector.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/dcs/ddc_service.c b/drivers/gpu/drm/amd/dal/dc/dcs/ddc_service.c
index 854ff3f..bbab51c 100644
--- a/drivers/gpu/drm/amd/dal/dc/dcs/ddc_service.c
+++ b/drivers/gpu/drm/amd/dal/dc/dcs/ddc_service.c
@@ -23,7 +23,8 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
+
 #include "include/adapter_service_interface.h"
 #include "include/i2caux_interface.h"
 #include "include/ddc_service_interface.h"
@@ -33,7 +34,6 @@
 #include "include/logger_interface.h"
 #include "ddc_i2caux_helper.h"
 #include "ddc_service.h"
-#include "dal_services_types.h"
 
 #define AUX_POWER_UP_WA_DELAY 500
 #define I2C_OVER_AUX_DEFER_WA_DELAY 70
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_ddc_dce110.c b/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_ddc_dce110.c
index f026464..b8554aa 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_ddc_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_ddc_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_factory_dce110.c b/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_factory_dce110.c
index 55d6986..6a9ee1a 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_factory_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_factory_dce110.c
@@ -27,7 +27,7 @@
  * Pre-requisites: headers required by header of this unit
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/gpio_types.h"
 #include "../hw_factory.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_hpd_dce110.c b/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_hpd_dce110.c
index 34405e9..e427f2c 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_hpd_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_hpd_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_translate_dce110.c b/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_translate_dce110.c
index 38512fa..0c87515 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_translate_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/dce110/hw_translate_dce110.c
@@ -27,7 +27,7 @@
  * Pre-requisites: headers required by header of this unit
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/gpio_types.h"
 #include "../hw_translate.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/ddc.c b/drivers/gpu/drm/amd/dal/dc/gpio/ddc.c
index 548b1cf..ee6a0b0 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/ddc.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/ddc.c
@@ -27,7 +27,7 @@
  * Pre-requisites: headers required by header of this unit
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/gpio_interface.h"
 #include "include/ddc_interface.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_ddc_diag.c b/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_ddc_diag.c
index f566241..282f1fc 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_ddc_diag.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_ddc_diag.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_factory_diag.c b/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_factory_diag.c
index 59503c4..2349bf7 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_factory_diag.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_factory_diag.c
@@ -27,7 +27,7 @@
  * Pre-requisites: headers required by header of this unit
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/gpio_types.h"
 #include "../hw_factory.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_hpd_diag.c b/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_hpd_diag.c
index baf5caf..f47e56a 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_hpd_diag.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_hpd_diag.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_translate_diag.c b/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_translate_diag.c
index 6d4da30..cfcd197 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_translate_diag.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/diagnostics/hw_translate_diag.c
@@ -24,7 +24,7 @@
  */
 
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/gpio_types.h"
 
 #include "../hw_translate.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/gpio_base.c b/drivers/gpu/drm/amd/dal/dc/gpio/gpio_base.c
index 6115f59..9168b2c 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/gpio_base.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/gpio_base.c
@@ -27,7 +27,7 @@
  * Pre-requisites: headers required by header of this unit
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/gpio_interface.h"
 #include "include/gpio_service_interface.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/gpio_service.c b/drivers/gpu/drm/amd/dal/dc/gpio/gpio_service.c
index 0920545..b79653c 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/gpio_service.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/gpio_service.c
@@ -27,7 +27,7 @@
  * Pre-requisites: headers required by header of this unit
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/gpio_interface.h"
 #include "include/ddc_interface.h"
 #include "include/irq_interface.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/hw_ddc.c b/drivers/gpu/drm/amd/dal/dc/gpio/hw_ddc.c
index 0608f16..e15f3a2 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/hw_ddc.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/hw_ddc.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/hw_factory.c b/drivers/gpu/drm/amd/dal/dc/gpio/hw_factory.c
index 17b5fdf..a5fa3aa 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/hw_factory.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/hw_factory.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio.c b/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio.c
index 2964d5d..9e231d3 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio_pad.c b/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio_pad.c
index 057c439..52757ac 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio_pad.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio_pad.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio_pin.c b/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio_pin.c
index 4ab1848..0d3f07f 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio_pin.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/hw_gpio_pin.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/hw_hpd.c b/drivers/gpu/drm/amd/dal/dc/gpio/hw_hpd.c
index c09d74c..617d648 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/hw_hpd.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/hw_hpd.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/hw_translate.c b/drivers/gpu/drm/amd/dal/dc/gpio/hw_translate.c
index d22504f..4a894c8 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/hw_translate.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/hw_translate.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpio/irq.c b/drivers/gpu/drm/amd/dal/dc/gpio/irq.c
index 382b89f..04bb69d 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpio/irq.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpio/irq.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/calc_pll_clock_source.c b/drivers/gpu/drm/amd/dal/dc/gpu/calc_pll_clock_source.c
index acc8838..4947d73 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/calc_pll_clock_source.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/calc_pll_clock_source.c
@@ -22,7 +22,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "calc_pll_clock_source.h"
 #include "include/bios_parser_interface.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/clock_source.c b/drivers/gpu/drm/amd/dal/dc/gpu/clock_source.c
index a2e618e..73804cd 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/clock_source.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/clock_source.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/adapter_service_interface.h"
 #include "include/bios_parser_interface.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.c b/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.c
index f124dba..0ed4f06 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "dc_clock_generator.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/dce110/dc_clock_gating_dce110.c b/drivers/gpu/drm/amd/dal/dc/gpu/dce110/dc_clock_gating_dce110.c
index e2d4228..6edb5aa 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/dce110/dc_clock_gating_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/dce110/dc_clock_gating_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/logger_interface.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/dce110/display_clock_dce110.c b/drivers/gpu/drm/amd/dal/dc/gpu/dce110/display_clock_dce110.c
index 6cf3955..3e0e9b3 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/dce110/display_clock_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/dce110/display_clock_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "dce/dce_11_0_d.h"
 #include "dce/dce_11_0_sh_mask.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/dce110/ext_clock_source_dce110.c b/drivers/gpu/drm/amd/dal/dc/gpu/dce110/ext_clock_source_dce110.c
index ae70e41..fa3201b 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/dce110/ext_clock_source_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/dce110/ext_clock_source_dce110.c
@@ -22,7 +22,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "dce/dce_11_0_d.h"
 #include "dce/dce_11_0_sh_mask.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/dce110/pll_clock_source_dce110.c b/drivers/gpu/drm/amd/dal/dc/gpu/dce110/pll_clock_source_dce110.c
index 019ea02..ba05597 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/dce110/pll_clock_source_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/dce110/pll_clock_source_dce110.c
@@ -22,7 +22,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "dce/dce_11_0_d.h"
 #include "dce/dce_11_0_sh_mask.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/dce110/vce_clock_source_dce110.c b/drivers/gpu/drm/amd/dal/dc/gpu/dce110/vce_clock_source_dce110.c
index 0a0b516..249720f 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/dce110/vce_clock_source_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/dce110/vce_clock_source_dce110.c
@@ -21,7 +21,7 @@
  * Authors: AMD
  *
  */
-#include "dal_services.h"
+#include "dc_services.h"
 #include "vce_clock_source_dce110.h"
 #include "include/clock_source_types.h"
 #include "include/bios_parser_interface.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/display_clock.c b/drivers/gpu/drm/amd/dal/dc/gpu/display_clock.c
index a11aa84..887bd74 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/display_clock.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/display_clock.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "display_clock.h"
 #include "adapter_service_interface.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/divider_range.c b/drivers/gpu/drm/amd/dal/dc/gpu/divider_range.c
index 3b04447..b4355f2 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/divider_range.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/divider_range.c
@@ -22,7 +22,7 @@
  * Authors: AMD
  *
  */
-#include "dal_services.h"
+#include "dc_services.h"
 #include "divider_range.h"
 
 bool dal_divider_range_construct(
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/ext_clock_source.c b/drivers/gpu/drm/amd/dal/dc/gpu/ext_clock_source.c
index 6cd0b3f..ec5b17d 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/ext_clock_source.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/ext_clock_source.c
@@ -22,7 +22,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/bios_parser_interface.h"
 #include "include/clock_source_types.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/pll_clock_source.c b/drivers/gpu/drm/amd/dal/dc/gpu/pll_clock_source.c
index aa5a667..d00bb61 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/pll_clock_source.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/pll_clock_source.c
@@ -22,7 +22,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/bios_parser_interface.h"
 #include "pll_clock_source.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/aux_engine.c b/drivers/gpu/drm/amd/dal/dc/i2caux/aux_engine.c
index 824ceec..b81fbdb 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/aux_engine.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/aux_engine.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
index 1b40a78..d0b8288 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_hw_engine_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_hw_engine_dce110.c
index 17e89ce..ce3cc4d 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_hw_engine_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_hw_engine_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 #include "include/logger_interface.h"
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_sw_engine_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_sw_engine_dce110.c
index c415a4e..2d5a318 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_sw_engine_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2c_sw_engine_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2caux_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2caux_dce110.c
index 05f5778..d2de0f2 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2caux_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/i2caux_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/diagnostics/i2caux_diag.c b/drivers/gpu/drm/amd/dal/dc/i2caux/diagnostics/i2caux_diag.c
index 86b606d..96b78e7 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/diagnostics/i2caux_diag.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/diagnostics/i2caux_diag.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/engine_base.c b/drivers/gpu/drm/amd/dal/dc/i2caux/engine_base.c
index 2f87a65..7a1c78c 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/engine_base.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/engine_base.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_engine.c b/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_engine.c
index 78c7d61..c8ab1f8 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_engine.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_engine.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_generic_hw_engine.c b/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_generic_hw_engine.c
index d91e259..61df97e 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_generic_hw_engine.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_generic_hw_engine.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_hw_engine.c b/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_hw_engine.c
index 77f2b84..cd8aa44 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_hw_engine.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_hw_engine.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_sw_engine.c b/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_sw_engine.c
index c253917..c14c5df 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_sw_engine.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/i2c_sw_engine.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c b/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c
index 68409b3..68dff0e 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 /*
  * Pre-requisites: headers required by header of this unit
diff --git a/drivers/gpu/drm/amd/dal/dc/inc/core_dc.h b/drivers/gpu/drm/amd/dal/dc/inc/core_dc.h
index 2d5644c..d794132 100644
--- a/drivers/gpu/drm/amd/dal/dc/inc/core_dc.h
+++ b/drivers/gpu/drm/amd/dal/dc/inc/core_dc.h
@@ -10,7 +10,6 @@
 
 #include "core_types.h"
 #include "hw_sequencer.h"
-#include "dal_services.h"
 
 struct dc {
 	struct dc_context *ctx;
diff --git a/drivers/gpu/drm/amd/dal/dc/inc/link_encoder.h b/drivers/gpu/drm/amd/dal/dc/inc/link_encoder.h
index df9019a..ab99b27 100644
--- a/drivers/gpu/drm/amd/dal/dc/inc/link_encoder.h
+++ b/drivers/gpu/drm/amd/dal/dc/inc/link_encoder.h
@@ -8,7 +8,6 @@
 #ifndef LINK_ENCODER_H_
 #define LINK_ENCODER_H_
 
-#include "dal_services_types.h"
 #include "grph_object_defs.h"
 #include "signal_types.h"
 #include "dc_types.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/irq/dce110/irq_service_dce110.c b/drivers/gpu/drm/amd/dal/dc/irq/dce110/irq_service_dce110.c
index 2a4f14c..3caeeed 100644
--- a/drivers/gpu/drm/amd/dal/dc/irq/dce110/irq_service_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/irq/dce110/irq_service_dce110.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/logger_interface.h"
 
diff --git a/drivers/gpu/drm/amd/dal/dc/irq/irq_service.c b/drivers/gpu/drm/amd/dal/dc/irq/irq_service.c
index b54e813..6f625dd 100644
--- a/drivers/gpu/drm/amd/dal/dc/irq/irq_service.c
+++ b/drivers/gpu/drm/amd/dal/dc/irq/irq_service.c
@@ -23,7 +23,7 @@
  *
  */
 
-#include "dal_services.h"
+#include "dc_services.h"
 
 #include "include/irq_service_interface.h"
 #include "include/logger_interface.h"
diff --git a/drivers/gpu/drm/amd/dal/dc/irq_types.h b/drivers/gpu/drm/amd/dal/dc/irq_types.h
index f8f2395..35a0991 100644
--- a/drivers/gpu/drm/amd/dal/dc/irq_types.h
+++ b/drivers/gpu/drm/amd/dal/dc/irq_types.h
@@ -26,8 +26,6 @@
 #ifndef __DAL_IRQ_TYPES_H__
 #define __DAL_IRQ_TYPES_H__
 
-#include "dal_services_types.h"
-
 struct dc_context;
 
 typedef void (*interrupt_handler)(void *);
diff --git a/drivers/gpu/drm/amd/dal/dc/virtual/virtual_link_encoder.c b/drivers/gpu/drm/amd/dal/dc/virtual/virtual_link_encoder.c
index 4d48daa..4880341 100644
--- a/drivers/gpu/drm/amd/dal/dc/virtual/virtual_link_encoder.c
+++ b/drivers/gpu/drm/amd/dal/dc/virtual/virtual_link_encoder.c
@@ -23,6 +23,8 @@
  *
  */
 
+#include "dc_services_types.h"
+
 #include "virtual_link_encoder.h"
 
 #define VIRTUAL_MAX_PIXEL_CLK_IN_KHZ 600000
diff --git a/drivers/gpu/drm/amd/dal/dc/virtual/virtual_stream_encoder.c b/drivers/gpu/drm/amd/dal/dc/virtual/virtual_stream_encoder.c
index dcfda67..99784be 100644
--- a/drivers/gpu/drm/amd/dal/dc/virtual/virtual_stream_encoder.c
+++ b/drivers/gpu/drm/amd/dal/dc/virtual/virtual_stream_encoder.c
@@ -23,9 +23,10 @@
  *
  */
 
-#include "virtual_stream_encoder.h"
 #include "dc_services.h"
 
+#include "virtual_stream_encoder.h"
+
 static void virtual_stream_encoder_dp_set_stream_attribute(
 	struct stream_encoder *enc,
 	struct dc_crtc_timing *crtc_timing) {}
diff --git a/drivers/gpu/drm/amd/dal/include/dal_types.h b/drivers/gpu/drm/amd/dal/include/dal_types.h
index fe884da..3739776 100644
--- a/drivers/gpu/drm/amd/dal/include/dal_types.h
+++ b/drivers/gpu/drm/amd/dal/include/dal_types.h
@@ -26,7 +26,6 @@
 #ifndef __DAL_TYPES_H__
 #define __DAL_TYPES_H__
 
-#include "dal_services_types.h"
 #include "signal_types.h"
 #include "dc_types.h"
 
diff --git a/drivers/gpu/drm/amd/dal/include/fixed31_32.h b/drivers/gpu/drm/amd/dal/include/fixed31_32.h
index 4577809..507f9f6 100644
--- a/drivers/gpu/drm/amd/dal/include/fixed31_32.h
+++ b/drivers/gpu/drm/amd/dal/include/fixed31_32.h
@@ -26,8 +26,6 @@
 #ifndef __DAL_FIXED31_32_H__
 #define __DAL_FIXED31_32_H__
 
-#include "dal_services_types.h"
-
 /*
  * @brief
  * Arithmetic operations on real numbers
diff --git a/drivers/gpu/drm/amd/dal/include/fixed32_32.h b/drivers/gpu/drm/amd/dal/include/fixed32_32.h
index 5291a30..5fca957 100644
--- a/drivers/gpu/drm/amd/dal/include/fixed32_32.h
+++ b/drivers/gpu/drm/amd/dal/include/fixed32_32.h
@@ -26,9 +26,6 @@
 #ifndef __DAL_FIXED32_32_H__
 #define __DAL_FIXED32_32_H__
 
-#include "dal_services_types.h"
-
-
 struct fixed32_32 {
 	uint64_t value;
 };
diff --git a/drivers/gpu/drm/amd/dal/include/grph_object_id.h b/drivers/gpu/drm/amd/dal/include/grph_object_id.h
index 4938435..4c8079c 100644
--- a/drivers/gpu/drm/amd/dal/include/grph_object_id.h
+++ b/drivers/gpu/drm/amd/dal/include/grph_object_id.h
@@ -26,8 +26,6 @@
 #ifndef __DAL_GRPH_OBJECT_ID_H__
 #define __DAL_GRPH_OBJECT_ID_H__
 
-#include "dal_services_types.h"
-
 /* Types of graphics objects */
 enum object_type {
 	OBJECT_TYPE_UNKNOWN  = 0,
diff --git a/drivers/gpu/drm/amd/dal/include/link_service_types.h b/drivers/gpu/drm/amd/dal/include/link_service_types.h
index 30fc6f0..d2e6256 100644
--- a/drivers/gpu/drm/amd/dal/include/link_service_types.h
+++ b/drivers/gpu/drm/amd/dal/include/link_service_types.h
@@ -26,8 +26,6 @@
 #ifndef __DAL_LINK_SERVICE_TYPES_H__
 #define __DAL_LINK_SERVICE_TYPES_H__
 
-#include "dal_services_types.h"
-
 #include "grph_object_id.h"
 #include "dpcd_defs.h"
 #include "dal_types.h"
diff --git a/drivers/gpu/drm/amd/dal/include/signal_types.h b/drivers/gpu/drm/amd/dal/include/signal_types.h
index 827c316..a50f7ed 100644
--- a/drivers/gpu/drm/amd/dal/include/signal_types.h
+++ b/drivers/gpu/drm/amd/dal/include/signal_types.h
@@ -26,8 +26,6 @@
 #ifndef __DC_SIGNAL_TYPES_H__
 #define __DC_SIGNAL_TYPES_H__
 
-#include "dal_services_types.h"
-
 enum signal_type {
 	SIGNAL_TYPE_NONE		= 0L,		/* no signal */
 	SIGNAL_TYPE_DVI_SINGLE_LINK	= (1 << 0),
-- 
2.7.4

